haproxy/proxy.sls

```diff
@@ -77,4 +78,65 @@ haproxy_service:
   {%- endif %}
 {%- endfor %}
 
+
+
+{%- set _haproxy_xinetd_srv = [] %}
+
+{%- for listen_name, listen in proxy.get('listen', {}).iteritems() %}
+{%- if listen.type == 'mysql' and listen.get('checks', {}).get('clustercheck', {}).get('enabled', False) == True %}
+{%- for bind in listen.binds %}
+{%- set index = '_{0}_{1}'.format(bind.address, bind.port) %}
+{%- set _ccheck = listen.checks.clustercheck %}
+{%- do _haproxy_xinetd_srv.append('clustercheck') %}
+/etc/xinetd.d/mysql_clustercheck{{ index }}_{{ _ccheck.get('clustercheckport', 9200) }}:
+  file.managed:
+    - source: salt://haproxy/files/xinet.d.conf
+    - template: jinja
+    - defaults:
+        user: nobody
+        # FIXME, add optins if check_attr host/port is defined etc..
+        server: '/usr/local/bin/clustercheck {{ _ccheck.get('user', 'clustercheck') }} {{ _ccheck.get('password', 'clustercheck') }} {{ _ccheck.get('available_when_donor', 0) }} {{ _ccheck.get('available_when_readonly', 0) }}'
+        port: _ccheck.get('port', 9200)
+        flags: REUSE
+        per_source: UNLIMITED
+    - require:
+      - file: /usr/local/bin/mysql_clustercheck
+    - watch_in:
+      - haproxy_xinetd_service
+
+{%- endfor %}
+{%- endif %}
+{%- endfor %}
+
+{% if 'clustercheck' in _haproxy_xinetd_srv %}
+clustercheck_dir:
+  file.directory:
+  - name: /usr/local/bin/
+  - user: root
+  - group: root
+  - mode: 750
+
+/usr/local/bin/mysql_clustercheck:
+  file.managed:
+    - source: salt://haproxy/files/clustercheck.sh
+    - user: root
+    - group: root
+    - mode: 755
+    - require:
+      - file: clustercheck_dir
+{%- endif %}
+
+{%- if _haproxy_xinetd_srv|length > 0 %}
```

Comment:
now it really depends whether how to interpret haproxy cfg `check port 9200` is that port on haproxy node? locally or on `server listen 1.2.3.4` (aka remote)?

if local, it might be a reasonable implement some checks with xinetd locally